packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -348,3 +394,97 @@ class SourcePosition {
   @override
   String toString() => '$line $column';
 }
+
+/// A tuple of a breakpoint and a source position.
+class BreakpointAndSourcePosition
+    implements Comparable<BreakpointAndSourcePosition> {
+  BreakpointAndSourcePosition(this.breakpoint, [this.sourcePosition]);
+
+  final Breakpoint breakpoint;
+  final SourcePosition sourcePosition;
+
+  bool get resolved => breakpoint.resolved;
+
+  ScriptRef get script {
+    if (breakpoint.location is UnresolvedSourceLocation) {
+      final UnresolvedSourceLocation location = breakpoint.location;
```

Comment:
do we need the type? shouldn't type promotion do this automatically? same comment elsewhere